pkg/datapath/linux/probes/probes.go

```diff
@@ -386,10 +388,137 @@ func HaveBoundedLoops() error {
+			LinkAttrs: netlink.LinkAttrs{
+				Name: "dummy",
+			},
+		}
+		if err := netlink.LinkAdd(dummy); err != nil {
+			return fmt.Errorf("unable to create dummy iface: %w", err)
+		}
+
+		//Bring link up on lo&dummy
+		if err := netlink.LinkSetUp(lo); err != nil {
+			return fmt.Errorf("unable to bring up loopback iface: %w", err)
+		}
+		if err := netlink.LinkSetUp(dummy); err != nil {
+			return fmt.Errorf("unable to bring up dummy iface: %w", err)
+		}
+
+		//Addr
+		addrStr := "10.0.0.1/24"
+		addr, _ := netlink.ParseAddr(addrStr)
+		if err := netlink.AddrAdd(dummy, addr); err != nil {
+			return fmt.Errorf("unable to add addr %s to dummy iface: %w", addrStr, err)
+		}
+
+		//Add dummy unresolvable route
+		gwAddrStr := "10.0.0.2"
+		dstCIDR := "1.2.3.4/32"
+		_, dst, _ := net.ParseCIDR(dstCIDR)
+		route := &netlink.Route{
+			LinkIndex: dummy.Attrs().Index,
+			Dst:       dst,
+			Gw:        net.ParseIP(gwAddrStr),
+		}
+		if err := netlink.RouteAdd(route); err != nil {
+			return fmt.Errorf("unable to add route %s via %s dev dummy: %w", dstCIDR, gwAddrStr, err)
+		}
+
+		//netlink stores bytes in NBO, so use Native to conver to uint32
+		nhBe := binary.NativeEndian.Uint32(dst.IP.To4())
+
+		//Attach eBPF program to call bpf_fib_lookup()
+		probe_name := "have_fib_index_probe"
+		progSpec := &ebpf.ProgramSpec{
+			Type:    ebpf.SchedCLS,
+			License: "GPL",
+		}
+
+		progSpec.Instructions = asm.Instructions{
+			//Memset lookup struct
+			asm.Mov.Imm(asm.R2, 0),
+			asm.StoreMem(asm.RFP, -8, asm.R2, asm.DWord),
+			asm.StoreMem(asm.RFP, -16, asm.R2, asm.DWord),
+			asm.StoreMem(asm.RFP, -24, asm.R2, asm.DWord),
+			asm.StoreMem(asm.RFP, -32, asm.R2, asm.DWord),
+			asm.StoreMem(asm.RFP, -40, asm.R2, asm.DWord),
+			asm.StoreMem(asm.RFP, -48, asm.R2, asm.DWord),
+			asm.StoreMem(asm.RFP, -56, asm.R2, asm.DWord),
+			asm.StoreMem(asm.RFP, -64, asm.R2, asm.DWord),
+			asm.StoreMem(asm.RFP, -72, asm.R2, asm.DWord),
+			asm.StoreMem(asm.RFP, -80, asm.R2, asm.DWord),
+
+			//Set AF_INET
+			asm.Mov.Imm(asm.R2, 2),
+			asm.StoreMem(asm.RFP, -80, asm.R2, asm.DWord),
+
+			//Set dst IP (1.2.3.4)
+			asm.Mov.Imm(asm.R2, int32(nhBe)),
+			asm.StoreMem(asm.RFP, -48, asm.R2, asm.DWord),
+
+			//Set ifindex to lo (1)
+			asm.Mov.Imm(asm.R2, 1),
+			asm.StoreMem(asm.RFP, -72, asm.R2, asm.DWord),
+
+			//Args
+			//R1 (ctx/skbuf) unmodified
+			asm.Mov.Reg(asm.R2, asm.RFP), //struct bpf_fib_lookup
+			asm.Add.Imm(asm.R2, -80),
+			asm.Mov.Imm(asm.R3, 64),                             //len
+			asm.Mov.Imm(asm.R4, unix.BPF_FIB_LOOKUP_SKIP_NEIGH), //L2 NH not resolvable/don't care
+			asm.FnFibLookup.Call(),
+
+			//Recover ifindex to return
+			asm.LoadMem(asm.R0, asm.RFP, -72, asm.Word),
+			asm.Return(),
+		}
+
+		prog, err := ebpf.NewProgram(progSpec)
+		if err != nil {
+			return fmt.Errorf("unable to load eBPF program '%s': %w", probe_name, err)
+		}
+		defer prog.Close()
+
+		pkt := []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
```

Comment:
Slightly more expressive:
```suggestion
		pkt := make([]byte, 16)
```

Reply:
Existing probes do the same: https://github.com/cilium/cilium/blob/ab0a12a428cba40e7e662627ceee1c8d5881fd94/pkg/datapath/linux/probes/probes.go#L646

Also, do we care about performance on a probe?

Reply:
~Expressive~, not expensive. Fine by me, just don't like counting zeroes. :stuck_out_tongue:

Reply:
I will adopt the change, and change the other probe.